### PR TITLE
impr: update to dpp 0.12

### PR DESCRIPTION
**Pull request by Alex-Werner**
### Issue being fixed or implemented  

With the pending release of @dashevo/js-dpp version 0.12, we needed to update some of the breaking changes.

### What was done  

- fix: removed createDataContract
- fix: removed identity types (user, application)
- fix: properly release (throw) catched error 
- impr: used DPP factories for STCreation
- chore: bumped wallet-lib to 6.1
- fix: updated DPNS Contract
- breaking: identities.register now return the identity
- feat: contract.register now return the dataContract

### How Has This Been Tested?

- updated the fixtures

### Notes  

With changes in platform data on evonet, the new ID for our ratePlatform app is ATU1EJjKhWZzhmuRdCvuNbpos7nwpbDqtrZ3yvxxjpKL. 
Do not hesitate to submit your rating by referencing the app id to your SDK instance and creating a new rate document :)

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/SDK/Client/Platform/methods/identities/register.ts [hunk 1]
```diff
@@ -2,34 +2,29 @@ import {PublicKey, PrivateKey, Transaction} from "@dashevo/dashcore-lib";
...
-const Identity = require('@dashevo/dpp/lib/identity/Identity');
 const stateTransitionTypes = require('@dashevo/dpp/lib/stateTransition/stateTransitionTypes');
 const IdentityPublicKey = require('@dashevo/dpp/lib/identity/IdentityPublicKey');
 const IdentityCreateTransition = require('@dashevo/dpp/lib/identity/stateTransitions/identityCreateTransition/IdentityCreateTransition');
 
 import {Platform} from "../../Platform";
 
 /**
- * Register identities to the platform
+ * Register identities to the platformz
```
Comment (thephez) [inline src/SDK/Client/Platform/methods/identities/register.ts:12]:
```suggestion
 * Register identities to the platform
```

File: src/SDK/Client/Client.ts [hunk 2]
```diff
@@ -147,6 +147,7 @@ export class Client {
             })
             .catch((e) => {
                 console.error('SDK apps fetching : failed to init', e);
+                throw e;
```
Comment (shumkov) [inline src/SDK/Client/Client.ts:150]: Return e in case of promise?

**Comment by shumkov** [comment]
@thephez yeah, we are waiting for wallet lib 6.1.0

File: src/SDK/Client/Platform/methods/contracts/broadcast.ts [hunk 3]
```diff
@@ -5,5 +5,26 @@
...
+
+    // @ts-ignore
+    const { privateKey } = account.getIdentityHDKey(0);
+
+    dataContractCreateTransition.sign(
+        identity.getPublicKeyById(0),
+        privateKey,
+    );
+
+    const result = await this.dpp.stateTransition.validateStructure(dataContractCreateTransition);
```
Comment (thephez) [inline src/SDK/Client/Platform/methods/contracts/broadcast.ts:23]: Is `this` necessary? There is a mismatch between this and the validation in documents/broadcast.
```suggestion
    const result = await dpp.stateTransition.validateStructure(dataContractCreateTransition);
```

File: docs/platform/documents/broadcast.md [hunk 4]
```diff
@@ -8,4 +8,4 @@ Parameters:
 | **document**              | Document  | yes            | A valid [created document](/platform/documents/create.md) |
```
Comment (shumkov) [inline docs/platform/documents/broadcast.md:8]: Lets, describe documents object structure?

File: docs/platform/documents/broadcast.md [hunk 5]
```diff
@@ -8,4 +8,4 @@ Parameters:
...
 | **identity**              | Identity  | yes            | A valid [registered identity](/platform/identities/register.md) |
 
-Returns : void.
+Returns : document.
```
Comment (shumkov) [inline docs/platform/documents/broadcast.md:11]: What if i pass many docs?

File: src/SDK/Client/Platform/methods/documents/broadcast.ts [hunk 6]
```diff
@@ -1,9 +1,17 @@
...
  * @param identity - identity
  */
-import broadcast from "../broadcastRecords";
-export default broadcast;
+export default async function broadcast(this: Platform, documents: { create: any[], replace: any[], delete: any[]}, identity: any): Promise<any> {
+    const { dpp } = this;
+
+    const documentsBatchTransition = dpp.documents.createStateTransition(documents);
+
+    await broadcastStateTransition(this, documentsBatchTransition, identity);
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/documents/broadcast.ts:16]: In docs, you return docs. Does it make sense to return anything from here?

File: docs/platform/documents/broadcast.md (hunk 5 above)
Comment (Alex-Werner) [reply docs/platform/documents/broadcast.md:14]: Good question. Especially if there is some that succeed and some that fails. 
Should we returns documents{} input ? or we keep to void ?

File: src/SDK/Client/Platform/methods/documents/broadcast.ts (hunk 6 above)
Comment (Alex-Werner) [reply src/SDK/Client/Platform/methods/documents/broadcast.ts:19]: I want to return documents to not be void, but yeah, may be not best solution ?

File: docs/platform/documents/broadcast.md (hunk 4 above)
Comment (Alex-Werner) [reply docs/platform/documents/broadcast.md]: It has been outdated, could you recheck to tell me if that is fine ?

File: src/SDK/Client/Platform/methods/documents/broadcast.ts (hunk 6 above)
Comment (shumkov) [reply src/SDK/Client/Platform/methods/documents/broadcast.ts:19]: I would return void from broadcast methods for now and later figure out what is most convenient thing can be for users.

Comment (Alex-Werner) [reply src/SDK/Client/Platform/methods/documents/broadcast.ts:19]: I did the opposite by just returning documents.

**Pull request merged**